File: Code/Aula_05/Ex4.c
#include <stdio.h>
#include <fcntl.h>
#include <unistd.h>
#include <stdlib.h>

int main(int argc, const char * argv[])
{
	int fp;
	
	fp = open("exercicio4.txt", O_RDWR | O_CREAT, S_IRWXU);
	if(fp==-1)
	{
		printf("Erro na abertura do arquivo.\n");
		exit(-1);
	}
	printf("\n");
	printf("Arquivo 'exercicio4.txt' criado. Detalhes:\n");
	system("ls -la exercicio4.txt");
	printf("Percebeu a diferenca na chamada de open()?\n");
	printf("\n");
	close(fp);

	return 0;
}

File: 2_Programação_sistemas_operacionais/2.2_File_POSIX/Python/Ex4.py
import os, stat

try:
	fd = os.open("exercicio4.txt", os.O_RDWR | os.O_CREAT, stat.S_IRWXU)
except OSError:
	print("Erro na abertura do arquivo.")
	exit(-1)
print("Arquivo 'exercicio4.txt' criado. Detalhes:")
os.system("ls -l exercicio4.txt")
os.close(fd)